lib/ansible/modules/cloud/vmware/vmware_vspan_session.py

```diff
@@ -0,0 +1,582 @@
+            for dv_port in dv_ports:
+                if dv_port.config.setting.securityPolicy.allowPromiscuous.value:
+                    self.set_port_security_promiscuous(dv_port.key, False)
+                    self.modified_ports.update({dv_port.key: True})
+                    promiscuous_ports.append(dv_port.key)
+        if ports_of_selected_session:
+            current_dv_ports = self.dv_switch.FetchDVPorts(vim.dvs.PortCriteria(portKey=ports_of_selected_session))
+            for dv_port in current_dv_ports:
+                if dv_port.config.setting.securityPolicy.allowPromiscuous.value:
+                    self.set_port_security_promiscuous(dv_port.key, False)
+                    self.modified_ports.update({dv_port.key: True})
+        # Return the promiscuous ports array, to set them back after the config is finished.
+        return promiscuous_ports
+
+    def delete_mirroring_session(self, key):
+        """Deletes the mirroring session.
+
+        Parameters
+        ----------
+        key : str
+            Key of the Session
+        """
+        session = vim.dvs.VmwareDistributedVirtualSwitch.VspanSession(
+            key=key
+        )
+        config_version = self.dv_switch.config.configVersion
+        s_spec = vim.dvs.VmwareDistributedVirtualSwitch.VspanConfigSpec(vspanSession=session, operation="remove")
+        c_spec = vim.dvs.VmwareDistributedVirtualSwitch.ConfigSpec(vspanConfigSpec=[s_spec], configVersion=config_version)
+        task = self.dv_switch.ReconfigureDvs_Task(c_spec)
+        try:
+            wait_for_task(task)
+        except Exception:
+            self.restore_original_state()
+            self.module.fail_json(msg=task.info.error.msg)
+
+    def restore_original_state(self):
+        """In case of failure restore, the changes we made."""
+        for port, state in self.modified_ports.items():
+            self.set_port_security_promiscuous(port, state)
+        if self.deleted_session is not None:
+            session = self.deleted_session
+            config_version = self.dv_switch.config.configVersion
+            s_spec = vim.dvs.VmwareDistributedVirtualSwitch.VspanConfigSpec(vspanSession=session, operation="add")
+            c_spec = vim.dvs.VmwareDistributedVirtualSwitch.ConfigSpec(vspanConfigSpec=[s_spec], configVersion=config_version)
+            # Revert the delete
+            task = self.dv_switch.ReconfigureDvs_Task(c_spec)
+            try:
+                wait_for_task(task)
+            except Exception:
+                self.restore_original_state()
+                self.module.fail_json(msg=task.info.error.msg)
+
+    def remove_vspan_session(self):
+        """Calls the necessary functions to delete a VSpanSession."""
+        results = dict(changed=False, result="")
+        mirror_session = self.find_session_by_name()
+        if mirror_session is None:
+            self.module.fail_json(msg='There is no VSpanSession with this name.')
```

Comment:
I suggest not failing, just return a `changed=False` result and exit the module successful, as this indicates that the session was already removed. Which is the expected result 
```suggestion
            self.module.fail_json(msg="There is no VSpanSession with the name: {0:s}.".format(self.name))
```